src/filters/examples/filter-panel.js

```diff
@@ -0,0 +1,363 @@
+                filter.values.forEach(function (value) {
+                  if(value.selected) {
+                    filterValues.push(value.value)
+                  }
+                });
+                if (filterValues.length > 0) {
+                  newAppliedFilters.push( createAppliedFilter (filter, filterValues) );
+                }
+              } else {
+                // each keyword value gets a new appliedFilter
+                filter.values.forEach(function (value) {
+                  filterValues = [value];
+                  newAppliedFilters.push( createAppliedFilter (filter, filterValues) );
+                });
+              }
+            }
+          });
+
+          // sets the filter result tags
+          $scope.filterConfig.appliedFilters = newAppliedFilters;
+          // filter items
+          filterItems($scope.filterConfig.appliedFilters);
+        };
+
+        var createAppliedFilter = function(filter, values) {
+          return {
+                    id: filter.id,
+                    title: filter.title,
+                    values: values
+                 };
+        }
+
+        var updateFilterPanelModel = function (changedFilterId, changedFilterValue) {
+          var changedFilter = _.find($scope.filterPanelModel, function(f) { return f.id === changedFilterId});
+          if(changedFilter.filterType === "checkbox") {
+             // unselect the checkbox
+             _.find(changedFilter.values, function(v) {return v.value == changedFilterValue}).selected = false;
+          } else {
+             // remove keyword from values array
+             _.remove(changedFilter.values, function(v) {return v == changedFilterValue});
+          }
+        };
+
+        var resetFilterPanelModel = function () {
+          $scope.filterPanelModel.forEach(function (filter) {
+            if (angular.isDefined(filter.values) && filter.values.length > 0) {
+              if(filter.filterType === "checkbox") {
+                // unselect all checkboxes
+                filter.values.forEach(function (value) {
+                  value.selected = false;
+                });
+              } else {
+                // clear all keyword filter values
+                filter.values = [];
+              }
+            }
+            });
+        };
+
+        var filterItems = function (filters) {
+          $scope.items = [];
+          if (filters && filters.length > 0) {
+            $scope.allItems.forEach(function (item) {
+              if (matchesFilters(item, filters)) {
+                $scope.items.push(item);
+              }
+            });
+          } else {
+            $scope.items = $scope.allItems;
+          }
+          $scope.filterConfig.resultsCount = $scope.items.length;
+        };
+
+        var matchesFilters = function (item, filters) {
+          var matches = true;
+
+          _.forEach(filters, function(filter) {
+              if (!matchesFilter(item, filter)) {
```

Comment:
[just gonna leave this here :trollface:](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)